client/registry_login_test.go

```diff
@@ -0,0 +1,56 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+func TestRegistryLoginError(t *testing.T) {
+	client := &APIClient{
+		HTTPCli: newMockClient(errorMockResponse(http.StatusInternalServerError, "Server error")),
+	}
+	loginConfig := types.AuthConfig{}
+	_, err := client.RegistryLogin(context.Background(), &loginConfig)
+	if err == nil || !strings.Contains(err.Error(), "Server error") {
+		t.Fatalf("expected a Server Error, got %v", err)
+	}
+}
+
+func TestRegistryLogin(t *testing.T) {
+	expectedURL := "/auth"
+
+	httpClient := newMockClient(func(req *http.Request) (*http.Response, error) {
+		if !strings.HasPrefix(req.URL.Path, expectedURL) {
+			return nil, fmt.Errorf("Expected URL '%s', got '%s'", expectedURL, req.URL)
+		}
+		loginConfig := types.AuthConfig{
```

Comment:
I see that in the test you input AuthConfig in `client.RegistryLogin(context.Background(), &types.AuthConfig{Username: "user_name", Password: "123456"})`. Why do you construct a new one in this place?

And actually this function would return a struct named `AuthResponse`, which is defined : 
```
type AuthResponse struct {

	// An opaque token used to authenticate a user after a successful login
	IdentityToken string `json:"IdentityToken,omitempty"`

	// The status of the authentication
	// Required: true
	Status string `json:"Status"`
}
```

Please change this part. Thanks a lot.

Reply:
I see. I mixed the get request with post request again. Will change it ASAP.